server/lobbyconnection.py

```diff
@@ -166,6 +191,15 @@ async def on_message_received(self, message):
                 'command': 'authentication_failed',
                 'text': ex.message
             })
+        except BanError as ex:
+            await self.send({
+                'command': 'notice',
+                'style': 'error',
+                'text': ex.message(),
+                'ban_reason': ex.ban_reason,
+                'ban_expiry': ex.ban_expiry.isoformat()
```

Comment:
It's a bit weird attaching these fields to the `notice` message, and the way that the client is written does not lend its self well to having dynamic messages (every message gets deserialized into a class, so the class needs to have members defined for all expected fields). So IMO, it's probably better to hold off on this until we add a dedicated ban message, otherwise if someone tries to implement it, it will turn into a mess.